flow/src/main/java/com/sequenceiq/flow/repository/FlowChainLogRepository.java

```diff
@@ -28,4 +32,22 @@ public interface FlowChainLogRepository extends CrudRepository<FlowChainLog, Lon
             + "WHERE fch.flowChainId NOT IN ( SELECT DISTINCT fl.flowChainId FROM FlowLog fl )"
             + " AND fch.flowChainId NOT IN (SELECT DISTINCT fc.parentFlowChainId FROM FlowChainLog fc)")
     int purgeOrphanFlowChainLogs();
+
+    @Query(value = "SELECT fcl.* " +
+            "FROM FlowChainLog fcl " +
+            "INNER JOIN ( " +
+            "    SELECT flowChainId, max(created) as created " +
+            "    FROM FlowChainLog " +
+            "    WHERE flowChainId IN (:flowChainIds) " +
+            "    GROUP by flowChainId " +
+            ")fch on fcl.flowChainId = fch.flowChainId and fcl.created = fch.created", nativeQuery = true,
+            countQuery = "SELECT count(fcl.*) " +
+                    "FROM FlowChainLog fcl " +
+                    "INNER JOIN ( " +
+                    "    SELECT flowChainId, max(created) as created " +
+                    "    FROM FlowChainLog " +
+                    "    WHERE flowChainId IN (:flowChainIds) " +
+                    "    GROUP by flowChainId " +
+                    ")fch on fcl.flowChainId = fch.flowChainId and fcl.created = fch.created")
+    Page<FlowChainLog> nativeFindByFlowChainIdInOrderByCreatedDesc(@Param("flowChainIds") Set<String> flowChainIds, Pageable pageable);
```

Comment:
tests are missing for this . We should use DataJpaTest which can also verify the paging.

Reply:
This is a native query and we cannot add @DataJpa test for native queries.

Reply:
Thanks. I missed that.